### ISSUE TITLE
RecursionError caused by method call in invariant

### ISSUE THREAD

**Issue by treiher** [Parquery/icontract#133]
Accessing a property or calling a class method leads to a RecursionError. Is it only supported to access attributes in the invariant?

```Python
from typing import Sequence
from icontract import invariant


@invariant(lambda self: all(t > 0 for t in self.numbers))
class Test:
    def __init__(self, numbers: Sequence[int]) -> None:
        self.__numbers = numbers

    @property
    def numbers(self) -> Sequence[int]:
        return self.__numbers


Test([])
```

```Console
Traceback (most recent call last):
  File "test.py", line 15, in <module>
    Test([])
  File "/home/tr/.local/lib/python3.8/site-packages/icontract/_checkers.py", line 435, in wrapper
    _assert_invariant(contract=contract, instance=instance)
  File "/home/tr/.local/lib/python3.8/site-packages/icontract/_checkers.py", line 160, in _assert_invariant
    check = contract.condition(self=instance)
[...]
  File "test.py", line 5, in <lambda>
    @invariant(lambda self: all(t > 0 for t in self.numbers))
  File "/home/tr/.local/lib/python3.8/site-packages/icontract/_checkers.py", line 445, in wrapper
    _assert_invariant(contract=contract, instance=instance)
  File "/home/tr/.local/lib/python3.8/site-packages/icontract/_checkers.py", line 160, in _assert_invariant
    check = contract.condition(self=instance)
  File "test.py", line 5, in <lambda>
    @invariant(lambda self: all(t > 0 for t in self.numbers))
  File "/home/tr/.local/lib/python3.8/site-packages/icontract/_checkers.py", line 442, in wrapper
    instance = _find_self(param_names=param_names, args=args, kwargs=kwargs)
  File "/home/tr/.local/lib/python3.8/site-packages/icontract/_checkers.py", line 402, in _find_self
    instance_i = param_names.index("self")
RecursionError: maximum recursion depth exceeded while calling a Python object
```

### Versions

icontract 2.2.0  
asttokens 2.0.4    
Python 3.8.2

**Comment by mristin** [comment]
Hi @treiher 
We have fixed it, see #129. It's pending review.

**Comment by mristin** [comment]
@treiher this has been fixed and the minor version has been bumped (#129 , release version 2.3.0).

@Sasha93 this issue can be closed.

**Comment by Sasha93** [comment]
Issue closed since issue was already fixed in release version 2.3.0

**Comment by treiher** [comment]
I can confirm that the issue is fixed in 2.3.0. Thanks!
